### ISSUE TITLE
 Using pytest.ini in a location other than where the tests are created results in the tests not being executed

### ISSUE THREAD

**Issue by MoElleuch** [DetachHead/pytest-robotframework#172]
I am developing my custom test runner, and it functions correctly. However, I encounter an issue after installing pytest_robotframework where the tests do not run following the test collections. I suspect there might be a conflict between the pytest_robotframework plugin and my pytest.ini file.

![sc](https://github.com/DetachHead/pytest-robotframework/assets/153929104/de27a643-ac4d-4dff-88b0-122a479745f2)

this is my pytest.ini :
```ini
[pytest]
python_files = test_*.py *_test.py tca_*.py
python_classes = Test* tca_*

log_cli = True
log_cli_level = Debug
log_cli_format = [%(asctime)s.%(msecs)03d] [%(levelname)-3s] [%(name)s] %(message)s
log_cli_date_format = %Y-%m-%d %H:%M:%S

log_file_level = Debug
log_file_format = [%(asctime)s.%(msecs)03d] [%(levelname)-3s] [%(name)s] %(message)s
log_file_date_format = %Y-%m-%d %H:%M:%S

junit_family = xunit1
junit_duration_report = call

addopts = -p pytest_robotframework
```

Is there any recommendation?

**Comment by DetachHead** [comment]
will be fixed in the next version :)

**Comment by MoElleuch** [comment]
@DetachHead Will version 2.5.2 support the fix? I've installed it, but the issue remains the same.

![image](https://github.com/DetachHead/pytest-robotframework/assets/153929104/7d65685c-955a-48de-89e0-8a38db328d06)

**Comment by DetachHead** [comment]
the fix worked for me and definitely in 2.5.2. could you please provide some more information about your setup:
- what's the command you're using to run pytest?
- where exactly is your `pytest.ini` file located in relation to your tests?

**Comment by MoElleuch** [comment]
exit_code = pytest.main(args, plugins)
This is the pytest args :
['C:\\new_xtr_ws\\xtr\\src\\xtr\\core\\runner\\pytest_plugins', 'C:\\new_xtr_ws\\demo-testing-projects\\pytest_project\\package1\\test\\test_tca1_1.py', 'C:\\new_xtr_ws\\demo-testing-projects\\pytest_project\\package1\\test\\test_tca1_2.py', '--junitxml=result\\last_execution\\Global\\xtr-results.xml', '-p', 'no:cacheprovider', '-v', '-s', '--alluredir=result\\last_execution\\Global\\allure-reports', '--timeout=300']

pytest.ini location : C:\\new_xtr_ws\\xtr\\src\\xtr\\core\\runner\\pytest_plugins 
tests path :
C:\\new_xtr_ws\\demo-testing-projects\\pytest_project\\package1\\test\\test_tca1_1.py
C:\\new_xtr_ws\\demo-testing-projects\\pytest_project\\package1\\test\\test_tca1_2.py

**Comment by DetachHead** [comment]
sorry but i still can't reproduce it. do you have any other plugins installed? could you maybe upload a git repo with a minimal reproduction of the issue?

**Comment by MoElleuch** [comment]
Sorry, I don't have a git repository for my project.
 Here are the plugins: allure-pytest-2.13.2, cov-3.0.0, pytest_robotframework-2.5.2, typeguard-4.1.5

![image](https://github.com/DetachHead/pytest-robotframework/assets/153929104/1418d789-6ea4-4ef4-bea2-8c1378593490)

And also, I defined a plugin with a pytest hook, but I tried running the project without my plugin, and I still have the same issue.

**Comment by DetachHead** [comment]
does it still happen if you disable all those plugins? does it work if you put the tests in the same location as your `pytest.ini`?

**Comment by MoElleuch** [comment]
- Disabling pytest_robotframework allows pytest to collect all items and execute them accordingly. 
- Disabling all other plugins except pytest-robotframework can lead to this error ! 

for pytest-robotframework 
**version 2.5.1** :
If the 'test' folder is placed in the same location as my pytest.ini file, pytest works perfectly. (However, I aim to define tests in a different location)
![image](https://github.com/DetachHead/pytest-robotframework/assets/153929104/efe8ce5b-1a3d-4069-a2fc-6a01288a8acf)

**version 2.5.2** : 
If placing the 'test' folder and pytest.ini in the same location still doesn't resolve the issue of tests not running.

![image](https://github.com/DetachHead/pytest-robotframework/assets/153929104/6f56c328-1949-4ff1-a82f-db1d78573983)

**version 3.0.0** : 
If running my custom test runner is contributing to an exception in file xdist_utils under folder pytest_robotframework\_internal
in this function **is_xdist_master** : 

![image](https://github.com/DetachHead/pytest-robotframework/assets/153929104/b6792d45-d5bb-4118-9599-5eaae393d12e)

As a solution to address this exception : 

`def is_xdist_master(session: Session):
    try :
        return session.config.option.numprocesses is not None
    except :
        return  False`

Even after resolving the exception, the issue with tests not running persists.

![image](https://github.com/DetachHead/pytest-robotframework/assets/153929104/72312e9e-df26-4a89-8ed7-819ef4221c40)

**Comment by DetachHead** [comment]
ok so i finally reproduced it. the problem is that you have both your tests and your pytest config file in a separate location to the working directory where pytest is being run. fixed in version 3.0.1
